docs/getting-started/publish-infrastructure.md

```diff
@@ -0,0 +1,567 @@
+> information.
+
+```yaml
+apiVersion: apiextensions.crossplane.io/v1alpha1
+kind: Composition
+metadata:
+  name: mysqlinstances.azure.database.example.org
+  labels:
+    provider: azure
+spec:
+  writeConnectionSecretsToNamespace: crossplane-system
+  reclaimPolicy: Delete
+  from:
+    apiVersion: database.example.org/v1alpha1
+    kind: MySQLInstance
+  to:
+    - base:
+        apiVersion: azure.crossplane.io/v1alpha3
+        kind: ResourceGroup
+        spec:
+          reclaimPolicy: Delete
+          providerRef:
+            name: azure-provider
+    - base:
+        apiVersion: database.azure.crossplane.io/v1beta1
+        kind: MySQLServer
+        spec:
+          forProvider:
+            administratorLogin: myadmin
+            resourceGroupNameSelector:
+              matchControllerRef: true
+            location: West US 2
+            sslEnforcement: Disabled
+            version: "5.7"
+            sku:
+              tier: GeneralPurpose
+              capacity: 2
+              family: Gen5
+            storageProfile:
+              storageMB: 20480
+          writeConnectionSecretToRef:
+            name: crossplane-system
+          providerRef:
+            name: azure-provider
+          reclaimPolicy: Delete
+      patches:
+        - fromFieldPath: "metadata.uid"
+          toFieldPath: "spec.writeConnectionSecretToRef.name"
+          transforms:
+            - type: string
+              string:
+                fmt: "%s-mysql"
+        - fromFieldPath: "spec.storageGB"
+          toFieldPath: "spec.forProvider.storageProfile.storageMB"
+          transforms:
+            - type: math
+              math:
+                multiply: 1024
+      connectionDetails:
+        - fromConnectionSecretKey: username
+        - fromConnectionSecretKey: password
+        - fromConnectionSecretKey: endpoint
+    - base:
+        apiVersion: database.azure.crossplane.io/v1alpha3
+        kind: MySQLServerFirewallRule
+        spec:
+          forProvider:
+            serverNameSelector:
+              matchControllerRef: true
+            resourceGroupNameSelector:
+              matchControllerRef: true
+            properties:
+              startIpAddress: 0.0.0.0
+              endIpAddress: 0.0.0.0
+          reclaimPolicy: Delete
+          providerRef:
+            name: example
+      patches:
+        - fromFieldPath: "metadata.uid"
+          toFieldPath: "spec.writeConnectionSecretToRef.name"
+          transforms:
+            - type: string
+              string:
+                fmt: "%s-mysql"
+        - fromFieldPath: "spec.storageGB"
+          toFieldPath: "spec.forProvider.storageProfile.storageMB"
+          transforms:
+            - type: math
+              math:
+                multiply: 1024
+      connectionDetails:
+        - fromConnectionSecretKey: username
+        - fromConnectionSecretKey: password
+        - fromConnectionSecretKey: endpoint
```

Comment:
As far as I can see, you don't need `patches` or `connectionDetails`. In fact, it causes error to have `spec.writeConnectionSecretToRef.name` without namespace.

Reply:
It was a typo to have `name` instead of `namespace` in the base, I am not sure why we would need no patches here?

Reply:
I think there is a copy pasta bug here - all of the patches and connection details here apply to the MySQLServerFirewallRule, but it looks like they were intended to apply to the MySQLServer.

Reply:
Ah my apologies! The code snippet with this comment did not show whether it was with the `MySQLServerFirewallRule` or the `MySQLServer`, but I now see that the patch and connection details were duplicated :+1: